mozilla_version/gecko.py

```diff
@@ -333,6 +349,71 @@ def _compare(self, other):
     def _compare_version_type(self, other):
         return self.version_type.compare(other.version_type)
 
+    def bump(self, field):
+        """Bump the number defined `field`.
+
+        This function handles previous edge cases.
+
+        Returns:
+            A new GeckoVersion with the right field bumped and the following ones set to 0,
+            if they exist.
+
+        """
+        if field == 'build_number' and self.build_number is None:
+            raise ValueError('Cannot bump the build number if it is not already set')
+        elif field == 'major_number' and self.is_esr:
+            raise ValueError('Cannot predictably know the next major ESR number')
+        elif (
+            (
+                field in (
+                    'minor_number', 'patch_number', 'release_candidate_number', 'old_fourth_number'
+                ) and
+                any((self.is_nightly, self.is_aurora_or_devedition, self.is_beta))
+            ) or
+            field == 'beta_number' and any((self.is_nightly, self.is_aurora_or_devedition))
+        ):
+            raise ValueError('Version type "{}" cannot bump "{}"'.format(self.version_type, field))
+
+        bump_kwargs = self._create_bump_kwargs(field)
+
+        if bump_kwargs.get('build_number') == 0:
+            del bump_kwargs['build_number']
+        if bump_kwargs.get('beta_number') == 0:
+            if self.is_beta:
+                bump_kwargs['beta_number'] = 1
+            else:
+                del bump_kwargs['beta_number']
+
+        if not self.is_four_digit_scheme:
+            del bump_kwargs['old_fourth_number']
+            if bump_kwargs.get('minor_number') == 0 and bump_kwargs.get('patch_number') == 0:
+                del bump_kwargs['patch_number']
+
+        if self.is_four_digit_scheme:
+            if (
+                bump_kwargs.get('patch_number') == 0 and
+                bump_kwargs.get('old_fourth_number') in (0, None)
+            ):
+                del bump_kwargs['patch_number']
+                del bump_kwargs['old_fourth_number']
+            elif (
+                bump_kwargs.get('patch_number') is None and
+                bump_kwargs.get('old_fourth_number') is not None and
+                bump_kwargs.get('old_fourth_number') > 0
+            ):
+                bump_kwargs['patch_number'] = 0
+
+        if self.is_rapid_release_scheme:
+            del bump_kwargs['release_candidate_number']
+
+        bump_kwargs['is_nightly'] = self.is_nightly
+        bump_kwargs['is_aurora_or_devedition'] = self.is_aurora_or_devedition
+        bump_kwargs['is_esr'] = self.is_esr
+
+        print(bump_kwargs)
```

Comment:
Nit: remove this debugging print statement

Reply:
👍